cli/pack/ignore_test.go

```diff
@@ -0,0 +1,893 @@
+		pattern: "foo*",
+		matches: []string{
+			"foo_blabla",
+			"foo",
+			"foo*",
+			"foo?",
+		},
+		mismatches: []string{
+			"2foo",
+			"blabla_2foo",
+			"2foo_blabla",
+			"bla_2foo_bla",
+		},
+	},
+	"name_with_asterisk_between": {
+		pattern: "f*oo",
+		matches: []string{
+			"f2oo",
+			"foo",
+			"f*oo",
+			"f?oo",
+		},
+		mismatches: []string{
+			"foo2",
+			"blabla_foo2",
+			"foo2_blabla",
+			"bla_foo2_bla",
+		},
+	},
+	"name_with_range_basic": {
+		pattern: "f[n-p]o",
+		matches: []string{
+			"fno",
+			"foo",
+			"fpo",
+		},
+		mismatches: []string{
+			"f2o",
+			"fmo",
+			"fqo",
+			"f?o",
+			"blabla_foo",
+			"foo_blabla",
+			"bla_foo_bla",
+		},
+	},
+	"name_with_range_inverted": {
+		pattern: "f[^n-p]o",
+		matches: []string{
+			"f2o",
+			"fmo",
+			"fqo",
+			"f?o",
+		},
+		mismatches: []string{
+			"foo",
+			"fno",
+			"fpo",
+		},
+	},
+	"name_with_set_basic": {
+		pattern: "[fgm]oo",
+		matches: []string{
+			"foo",
+			"goo",
+			"moo",
+		},
+		mismatches: []string{
+			"zoo",
+			"ooo",
+			"?oo",
+			"blabla_foo",
+			"foo_blabla",
+			"bla_foo_bla",
+		},
+	},
+	"name_with_set_inverted": {
+		pattern: "[^fgm]oo",
+		matches: []string{
+			"zoo",
+			"ooo",
+			"?oo",
+		},
+		mismatches: []string{
+			"foo",
+			"goo",
+			"moo",
+			"blabla_zoo",
+			"zoo_blabla",
+			"bla_zoo_bla",
+		},
+	},
+}
+
+type ignorePatternCase struct {
+	pattern    string
+	matches    []string
+	mismatches []string
+	dirOnly    bool
+	isNegate   bool
+}
+
+type ignorePatternCases map[string]ignorePatternCase
+
+func checkIgnorePattern(t *testing.T, testCases ignorePatternCases) {
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			p, err := createIgnorePattern(tc.pattern, "")
+			assert.Nil(t, err)
+			assert.NotNil(t, p.re)
+			assert.Equal(t, tc.dirOnly, p.dirOnly)
+			assert.Equal(t, tc.isNegate, p.isNegate)
+			for _, s := range tc.matches {
+				assert.Truef(t, p.re.MatchString(s), "%q doesn't match %s", s, p.re.String())
+			}
+			for _, s := range tc.mismatches {
+				assert.False(t, p.re.MatchString(s), "%q matches %s", s, p.re.String())
+			}
+		})
+	}
+}
+
+// Prepare basic test set for createIgnorePattern function
+var ignorePatternBasicCases = transformMapValues(ignoreTestData_names,
+	func(td ignoreTestData) ignorePatternCase {
+		return ignorePatternCase{
+			pattern: td.pattern,
+			// Expand with some meaningful paths (td.matches itself has no path-item).
+			matches: append(td.matches,
+				"in_subdir/"+td.matches[0],
+				"in/deep/nested/subdir/"+td.matches[0],
+			),
+			mismatches: td.mismatches,
+			dirOnly:    false,
+			isNegate:   false,
+		}
+	})
+
+func Test_createIgnorePattern_basic(t *testing.T) {
+	checkIgnorePattern(t, ignorePatternBasicCases)
+}
+
+func Test_createIgnorePattern_negate(t *testing.T) {
+	toNegate := func(tc ignorePatternCase) ignorePatternCase {
+		result := tc
+		result.pattern = "!" + tc.pattern
+		result.isNegate = true
+		return result
```

Comment:
```suggestion
		tc.pattern = "!" + tc.pattern
		tc.isNegate = true
		return tc
```
Up to you, I just don't see a reason why we need an extra copy.

Reply:
I mentioned it in your [suggestion PR](https://github.com/tarantool/tt/pull/1119)

> I doubt that you mean that, but it seems this way we change the original test set at each transformation which requires us to keep in mind what is in origin at any time, keep test orders or something similar. In my opinion this approach complicates the readability and here "the less code" doesn't mean "the better".

Reply:
Here we are working with a copy, not an original test case.

Reply:
Oops, sorry, my bad. "There" I missed that the whole map copied and "here"... no excuses, just facepalm (

Reply:
Fixed.